src/commands/publish.ts

```diff
@@ -84,17 +84,21 @@ export async function publish(options: BeachballOptions) {
   const revParseSuccessful = currentBranch || currentHash;
   const inBranch = currentBranch && currentBranch !== 'HEAD';
   const hasHash = currentHash !== null;
-  if (inBranch) {
-    console.log(`git checkout ${currentBranch}`);
-    gitFailFast(['checkout', currentBranch!], { cwd });
-  } else if (hasHash) {
-    console.log(`Looks like the repo was detached from a branch`);
-    console.log(`git checkout ${currentHash}`);
-    gitFailFast(['checkout', currentHash!], { cwd });
-  }
+  try {
```

Comment:
I don't think this is the right approach. `gitFailFast` is intended to be used with critical operations that are blocking if they fail. It also sets `process.exitCode = 1` even if you catch the error, so the step is likely to show up as failed in a CI build anyway.

A better approach would be to switch to using `git()` *only* for the branch deletion operation and logging a warning if the command failed. (The prior `git checkout` operations should continue to use `gitFailFast` because it's possible that the build could have later steps that will fail if it's not on the expected branch.)

Reply:
updated this, also updated similar case in bump and push